storage/util_1.7.go

```diff
@@ -5,8 +5,19 @@ package storage
 import (
 	"io"
 	"net/http"
+	"os"
 )
 
-func setContentLengthFromLimitedReader(req *http.Request, lr *io.LimitedReader) {
+func setContentLengthFromLimitedReader(req *http.Request, lr *io.LimitedReader) error {
 	req.ContentLength = lr.N
+	return nil
+}
+
+func setContentLengthFromFile(req *http.Request, f *os.File) error {
```

Comment:
This func would better live in `client.go` file. This file is for code that would only work on certain go versions.